components/match2/commons/match_group_util.lua

```diff
@@ -208,39 +211,47 @@ Creates a match group structure from its match records. Returns a value of type
 MatchGroupUtil.types.MatchGroup.
 ]]
 function MatchGroupUtil.makeMatchGroup(matchRecords)
-	local matches = Array.map(
-		matchRecords,
-		require('Module:Brkts/WikiSpecific').matchFromRecord
-	)
+	local type = matchRecords[1] and matchRecords[1].match2bracketdata.type or 'matchlist'
+	return type == 'matchlist'
+		and MatchGroupUtil.makeMatchlistFromRecords(matchRecords)
+		or MatchGroupUtil.makeBracketFromRecords(matchRecords)
```

Comment:
```suggestion
	if type == 'matchlist' then
		return MatchGroupUtil.makeMatchlistFromRecords(matchRecords)
	elseif type == 'bracket' then
		return MatchGroupUtil.makeBracketFromRecords(matchRecords)
	end
	return nil
```

Reply:
Is that to deal with a type that is neither matchlist or bracket?

I switched the order instead, so if the type is neither matchlist or bracket then uses the matchlist processing.

Reply:
I would prefer my version because that makes the error unavoidable

Reply:
> I would prefer my version because that makes the error unavoidable

I'll change it to throw then. I don't like signaling the error by returning `nil` in this particular case, because there's no way for callers to reasonably handle lpdb data errors. Better to let it go to the global error handler.

Reply:
Sounds good